pkg/strategy/funding/strategy.go

```diff
@@ -0,0 +1,266 @@
+package funding
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/exchange/binance"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/sirupsen/logrus"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+const ID = "funding"
+
+var log = logrus.WithField("strategy", ID)
+
+func init() {
+	// Register the pointer of the strategy struct,
+	// so that bbgo knows what struct to be used to unmarshal the configs (YAML or JSON)
+	// Note: built-in strategies need to imported manually in the bbgo cmd package.
+	bbgo.RegisterStrategy(ID, &Strategy{})
+}
+
+type Strategy struct {
+	*bbgo.Notifiability
+
+	// These fields will be filled from the config file (it translates YAML to JSON)
+	Symbol      string           `json:"symbol"`
+	Market      types.Market     `json:"-"`
+	Quantity    fixedpoint.Value `json:"quantity,omitempty"`
+	MaxPosition fixedpoint.Value `json:"maxPosition"`
+
+	FundingRate *struct {
+		High          fixedpoint.Value `json:"high"`
+		Neutral       fixedpoint.Value `json:"neutral"`
+		DiffThreshold fixedpoint.Value `json:"diffThreshold"`
+	} `json:"fundingRate"`
+
+	SupportDetection []struct {
+		Interval types.Interval `json:"interval"`
+
+		// MovingAverageType is the moving average indicator type that we want to use,
+		// it could be SMA or EWMA
+		MovingAverageType string `json:"movingAverageType"`
+
+		// MovingAverageInterval is the interval of k-lines for the moving average indicator to calculate,
+		// it could be "1m", "5m", "1h" and so on.  note that, the moving averages are calculated from
+		// the k-line data we subscribed
+		MovingAverageInterval types.Interval `json:"movingAverageInterval"`
```

Comment:
There is a struct combining both Interval and Window,

take a look at `types.IntervalWindow` or `bbgo.MovingAverageSettings`, see if we can reuse them and make the code clean